src/main/java/org/axion/axion_api_gateway/UserSession.java

```diff
@@ -55,7 +56,7 @@ private UserDTO fetchUserFromAuthService(String token) {
                 .build();
 
         try (Response response = client.newCall(request).execute()) {
-            if (response.isSuccessful()) {
+            if (response.isSuccessful() && response.body() != null) {
```

Comment:
Additional check to get rid of warning about response.body().string() causing NullPointerException.